service/src/test/java/com/myhome/controllers/AmenityControllerTest.java

```diff
@@ -213,6 +222,152 @@ void shouldNotUpdateCommunityAmenityIfAmenityNotExists() {
+        .willReturn(Collections.singletonList(getAmenityBookingsResponse));
+
+    // when
+    ResponseEntity<List<GetAmenityBookingsResponse>> amenitiesBookings =
+        amenityController.getAmenitiesBookings(TEST_AMENITY_ID,
+            null,
+            null,
+            null);
+
+    // then
+    assertEquals(HttpStatus.OK, amenitiesBookings.getStatusCode());
+    assertNotNull(amenitiesBookings.getBody());
+    assertEquals(1, amenitiesBookings.getBody().size());
+  }
+
+  @Test
+  void shouldReturnBookingsWithStartDateRange() {
+    // given
+    AmenityBookingItem amenityBookingItem = new AmenityBookingItem();
+    amenityBookingItem.setAmenity(getTestAmenity());
+    GetAmenityBookingsResponse getAmenityBookingsResponse = new GetAmenityBookingsResponse();
+    getAmenityBookingsResponse.setAmenityId(TEST_AMENITY_ID);
+    LocalDateTime currentTime = LocalDateTime.now();
+
+    List<AmenityBookingItem> amenityBookingItems = Collections.singletonList(amenityBookingItem);
+    given(amenitySDJpaService.listAllAmenityBookings(TEST_AMENITY_ID,
+        currentTime,
+        null,
+        null))
+        .willReturn(amenityBookingItems);
+    given(amenityBookingItemApiMapper.amenityBookingToAmenityBookingsResponse(amenityBookingItems))
+        .willReturn(Collections.singletonList(getAmenityBookingsResponse));
+
+    // when
+    ResponseEntity<List<GetAmenityBookingsResponse>> amenitiesBookings =
+        amenityController.getAmenitiesBookings(TEST_AMENITY_ID,
+            currentTime,
+            null,
+            null);
+
+    // then
+    assertEquals(HttpStatus.OK, amenitiesBookings.getStatusCode());
+    assertNotNull(amenitiesBookings.getBody());
+    assertEquals(1, amenitiesBookings.getBody().size());
+  }
+
+  @Test
+  void shouldReturnBookingsWithEndDateRange() {
+    // given
+    AmenityBookingItem amenityBookingItem = new AmenityBookingItem();
+    amenityBookingItem.setAmenity(getTestAmenity());
+    GetAmenityBookingsResponse getAmenityBookingsResponse = new GetAmenityBookingsResponse();
+    getAmenityBookingsResponse.setAmenityId(TEST_AMENITY_ID);
+    LocalDateTime futureTime = LocalDateTime.now().plusWeeks(2);
+
+    List<AmenityBookingItem> amenityBookingItems = Collections.singletonList(amenityBookingItem);
+    given(amenitySDJpaService.listAllAmenityBookings(TEST_AMENITY_ID,
+        null,
+        futureTime,
+        null))
+        .willReturn(amenityBookingItems);
+    given(amenityBookingItemApiMapper.amenityBookingToAmenityBookingsResponse(amenityBookingItems))
+        .willReturn(Collections.singletonList(getAmenityBookingsResponse));
+
+    // when
+    ResponseEntity<List<GetAmenityBookingsResponse>> amenitiesBookings =
+        amenityController.getAmenitiesBookings(TEST_AMENITY_ID,
+            null,
+            futureTime,
+            null);
+
+    // then
+    assertEquals(HttpStatus.OK, amenitiesBookings.getStatusCode());
+    assertNotNull(amenitiesBookings.getBody());
```

Comment:
`amenitiesBookings.getBody()` can be assigned to variable (in test context) as it is used twice:
```
    assertEquals(HttpStatus.OK, amenitiesBookings.getStatusCode());
    // here
    assertNotNull(amenitiesBookings.getBody());
    assertEquals(1, amenitiesBookings.getBody().size());
```